src/linear_algebra/matrix.lean

```diff
@@ -316,3 +316,31 @@ end
 end vector_space
 
 end matrix
+
+/-- The natural equivalence between linear endomorphisms of finite free modules and square matrices
+is compatible with the algebra structures. -/
+def alg_equiv_matrix' {R : Type v} [comm_ring R] [decidable_eq n] :
+  module.End R (n → R) ≃ₐ[R] matrix n n R :=
+{ map_mul'  := matrix.comp_to_matrix_mul,
+  map_add'  := linear_equiv_matrix'.map_add,
+  commutes' := λ r, by { change (r • (linear_map.id : module.End R _)).to_matrix = r • 1,
+                         rw ←linear_map.to_matrix_id, refl, },
+  ..linear_equiv_matrix' }
```

Comment:
For reasons that I don't completely comprehend, I think this is now no longer the way to extend structures... @gebner we need to provide `to_fun` and `inv_fun` explicitly, don't we?

Reply:
It's okay if it compiles.  I believe it is only a problem inside the `{}`, i.e. you only need to add `to_fun := ...` if `commutes'` breaks in mysterious ways.

Reply:
Ok, thanks!

Reply:
This sounds like something I'll bump into sooner or later so thanks for raising it, even if I got away with it this time.